src/cli/func-api.ts

```diff
@@ -16,6 +17,44 @@ export class FuncAPI {
     return funcCliCommand(['version']);
   }
 
+  static invokeFunctionLocal(
+    id: string,
+    location: string,
+    contextType: string,
+    format: string,
+    source: string,
+    type: string,
+    data: string,
+  ): string {
+    if (id) {
+      return `func invoke --id ${id} -p ${location} --content-type ${contextType} -f ${format} --source ${source} --type ${type} --data ${data}`;
+    }
+    return `func invoke -p ${location} --content-type ${contextType} -f ${format} --source ${source} --type ${type} --data ${data}`;
+  }
+
+  static invokeFunctionRemote(
+    id: string,
+    namespace: string,
+    contextType: string,
+    format: string,
+    source: string,
+    type: string,
+    data: string,
+    url: string,
+    location: string,
+  ): string {
+    if (id?.trim() && url?.trim()) {
+      return `func invoke -p ${location} --id ${id} -t ${url} -n ${namespace} --content-type ${contextType} -f ${format} --source ${source} --type ${type} --data ${data}`;
+    }
+    if (id?.trim()) {
+      return `func invoke -p ${location} --id ${id} -n ${namespace} --content-type ${contextType} -f ${format} --source ${source} --type ${type} --data ${data}`;
+    }
+    if (url?.trim()) {
+      return `func invoke -p ${location} -t ${url} -n ${namespace} --content-type ${contextType} -f ${format} --source ${source} --type ${type} --data ${data}`;
+    }
+    return `func invoke -p ${location} -n ${namespace} --content-type ${contextType} -f ${format} --source ${source} --type ${type} --data ${data}`;
```

Comment:
Why don't you create an array with args and then do `return funcCliCommand(args);` like what you did in the other methods? By writing the commands as strings if in future you need to add a new flag you need to edit N rows.
It would be better to do something like 
```
args = [];
if (id) { args.push(id) }
....
return funcCliCommand(args);
```

Reply:
done